src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -155,7 +155,17 @@ public int getTotalUserCount(){
     for (Entity message : results.asList(FetchOptions.Builder.withLimit(1000))){
       users.add(message.getProperty("user"));
     }
-    System.out.println(users);
     return users.size();
   }
+
+  /** Returns the average message length. */
+  public double getAverageMessageLength(){
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
+    Integer sum = 0;
+    for (Entity message : results.asList(FetchOptions.Builder.withLimit(1000))){
+      sum += message.getProperty("text").toString().length();
+    }
+    return sum.doubleValue()/results.asList(FetchOptions.Builder.withLimit(1000)).size();
```

Comment:
what could go wrong here? :stuck_out_tongue_winking_eye:

Reply:
the goal is to avoid division by zero, it's okay for sum to be zero in weird cases (a bunch of empty message?) so we should check that divisor is not 0 instead.